### ISSUE TITLE
`used-before-assignment` false positive for import nodes under `elif` after `TYPE_CHECKING`

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#8437]
### Bug description

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    pass
elif 42:
    import math


print(math)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:5:5: W0125: Using a conditional statement with a constant value (using-constant-test)
a.py:9:6: E0601: Using variable 'math' before assignment (used-before-assignment)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)
```

### Expected behavior

************* Module a
a.py:5:5: W0125: Using a conditional statement with a constant value (using-constant-test)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)

### Pylint version

```shell
pylint 3.0.0a6
astroid 2.16.0dev0
Python 3.11.0
```

**Comment by jacobtylerwalls** [comment]
@zenlyj tagging you so you're aware of the issue while developing; this is the sort of thing we could merge quickly and backport to 2.17.x, if you're interested in contributing a fix.

**Comment by zenlyj** [comment]
thanks for creating this issue! I would like to work on this one, please assign it to me.
